src/core/labeling/qgspallabeling.cpp

```diff
@@ -1406,8 +1407,10 @@ QPixmap QgsPalLayerSettings::labelSettingsPreviewPixmap( const QgsPalLayerSettin
   context.setFlag( Qgis::RenderContextFlag::Antialiasing, true );
 
   QWidget *activeWindow = QApplication::activeWindow();
-  const double logicalDpiX = activeWindow && activeWindow->screen() ? activeWindow->screen()->logicalDotsPerInchX() : 96.0;
-  context.setScaleFactor( logicalDpiX / 25.4 );
+  const double physicalDpiX = activeWindow && activeWindow->screen() ? activeWindow->screen()->physicalDotsPerInchX() : 96.0;
```

Comment:
Any reason why you switched from logical to physical DPI?

Reply:
Because it didn't work otherwise :man_shrugging:  -- I'd get a ~2/3 size difference between what it should be vs what it was appearing as

I **think** maybe the switch to the new hidpi handling flags meant we shouldn't be considering logical DPI anywhere anymore... but I'm not confident.

Reply:
@nyalldawson , hmm, OK; it's a shame Qt documentation on this doesn't help.

Reply:
Yeah. I'm quite confused to be honest -- I'd have expected the physical dpi / logical dpi / device pixel ratio to have a direct relation to each other, but on my setup at least there's no easy conversion between them...